src/EnergyPlus/UnitarySystem.hh

```diff
@@ -451,6 +514,15 @@ namespace UnitarySystems {
 
         static void getUnitarySystemInput(EnergyPlusData &state, std::string const &Name, bool const ZoneEquipment, int const ZoneOAUnitNum);
 
+        void processInputSpec(EnergyPlusData &state,
+                              const UnitarySysInputSpec &input_data,
+                              int sysNum,
+                              bool &errorsFound,
+                              bool const ZoneEquipment,
+                              int const ZoneOAUnitNum);
```

Comment:
Split getUnitarySystemInputData into 2 parts. The processInputSpec function processes all inputs based on the data in input_data. Then it returns and finishes processing. The setupOutputVariable calls are in getUnitarySystemInputData since the report variables for different models may be different from UnitarySystem.

Reply:
:+1: I like where this is heading!